lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -6269,6 +6269,24 @@ LogicalResult RWProbeOp::verifyInnerRefs(hw::InnerRefNamespace &ns) {
   return checks(symOp.getTargetResult().getType(), symOp.getLoc());
 }
 
+LogicalResult XMRRefOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
+  auto *target = symbolTable.lookupNearestSymbolFrom(*this, getRefAttr());
+  if (!target)
+    return emitOpError("invalid symbol reference");
+
+  // TODO: Verify that the target's type matches the type of this op.
+  return success();
+}
+
+LogicalResult XMRDerefOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
+  auto *target = symbolTable.lookupNearestSymbolFrom(*this, getRefAttr());
+  if (!target)
+    return emitOpError("invalid symbol reference");
```

Comment:
Your idea, but yeah maybe good to check these resolve to HierPathOp's 👍 .  And some quick error tests in `errors.mlir` (again your idea).